### PR TITLE
Add faster Exponential

### PR DESCRIPTION
**Pull request by rohanbabbar04**
## Description
- Add a class `Exponential` with all methods.
- Update test in `test_scipy.py`.

## Checklist

- [x] Code style is correct (follows pylint and black guidelines)
- [x] Includes new or updated tests to cover the new feature
- [x] New features are properly documented (with an example if appropriate)

### CODE REVIEW

File: preliz/tests/test_scipy.py [hunk 1]
```diff
@@ -13,6 +13,7 @@
     Binomial,
     NegativeBinomial,
     Poisson,
+    Exponential,
```
Comment (aloctavodia) [inline preliz/tests/test_scipy.py:16]: I am adding the distributions in two groups first the continuous then the discrete. Within each group the order is alphabetical.

**Review by aloctavodia — CHANGES_REQUESTED** [review]
Thanks!

File: preliz/distributions/exponential.py [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        Compute the percent point function (PPF) at a given probability q.
+        """
+        q = np.asarray(q)
+        return nb_ppf(q, self.beta)
+
+    def logpdf(self, x):
+        """
+        Compute the log probability density function (log PDF) at a given point x.
+        """
+        x = np.asarray(x)
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply preliz/distributions/exponential.py]: Removed

File: preliz/distributions/exponential.py [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+    def _fit_mle(self, sample, **kwargs):
+        mean, _ = mean_and_std(sample)
+        self._update(1 / mean)
+
+
+@nb.njit
+def nb_pdf(x, lam):
+    x_lam = x * lam
+    return np.where(x < 0, 0, lam * np.exp(-x_lam))
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply preliz/distributions/exponential.py]: Yes, now it is np.exp(logpdf).

File: preliz/distributions/exponential.py [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+@nb.njit
+def nb_cdf(x, lam):
+    x_lam = lam * x
+    return np.where(x < 0, 0, 1 - np.exp(-x_lam))
+
+
+@nb.njit
+def nb_ppf(q, beta):
+    return np.where((0 <= q) & (q <= 1), -beta * np.log(1 - q), np.nan)
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply preliz/distributions/exponential.py]: Done

File: preliz/distributions/exponential.py [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+
+
+@nb.njit
+def nb_logpdf(x, lam):
+    return np.where(x < 0, -np.inf, np.log(lam) - lam * x)
+
+
+@nb.njit
+def nb_neg_logpdf(x, lam):
+    return (-np.log(lam) + lam * x).sum()
```
[the remark this answers is not in the document]
Comment (rohanbabbar04) [reply preliz/distributions/exponential.py]: Changed to use nb_logpdf

File: preliz/distributions/exponential.py [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+def nb_neg_logpdf(x, lam):
+    return (-nb_logpdf(x, lam)).sum()
+
+
+@nb.njit
+def nb_entropy(beta):
+    return 1 + np.log(beta)
+
+
+@nb.njit
```
Comment (aloctavodia) [inline preliz/distributions/exponential.py:182]: Move to the `special` module, so it can be reused

**Review by aloctavodia — APPROVED** [review]

**Pull request merged**